task/sql-tasks.js

```diff
@@ -281,7 +470,34 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        WITH productInfo AS (

```

Comment:
cte look a bit cumbersome here. Select from a subquery "maxValues" and joining everything you need to  "maxValues" would suffice in this case